finality-provider/config/opstackl2.go

```diff
@@ -11,8 +11,9 @@ import (
 )
 
 type OPStackL2Config struct {
-	OPStackL2RPCAddress     string `long:"opstackl2-rpc-address" description:"the rpc address of the op-stack-l2 node to connect to"`
-	OPFinalityGadgetAddress string `long:"op-finality-gadget" description:"the contract address of the op-finality-gadget"`
+	OPStackL2RPCAddress      string `long:"opstackl2-rpc-address" description:"the rpc address of the op-stack-l2 node to connect to"`
+	OPFinalityGadgetAddress  string `long:"op-finality-gadget" description:"the contract address of the op-finality-gadget"`
```

Comment:
iiuc after we use the gadget RPC, we don't need this contract address anymore. wdyt?

note: removing it can be in a separate PR

Reply:
given `OPFinalityGadgetAddress` is still needed by the finality gadget server, i think it would be better to keep it here and avoid having to store it elsewhere + update the function signatures

Reply:
sure we can keep it for now. I was mostly thinking about that FG for different consumer chains will have diff RPC URLs and each consumer chain will have its own gadget address, which will be configured when starting the FP server. wdyt?